tests/unit/test_download.py

```diff
@@ -325,3 +325,18 @@ def test_insecure_host_cache_is_not_enabled(self, tmpdir):
         )
 
         assert not hasattr(session.adapters["https://example.com/"], "cache")
+
+
+def test_parse_credentials():
+    auth = MultiDomainBasicAuth()
+    assert auth.parse_credentials(u"foo:bar@example.com") == (u'foo', u'bar')
```

Comment:
In python 3, strings are natively unicode anyway; in python 2, I don't think this should be unicode here (either expecting to receive it, nor outputting). What you've output in the new function is unicode but that's because of the utf-8 jig I'm not sure about.